### PR TITLE
feat: adds support to list sites #345

### PR DESCRIPTION
**Pull request by srinandan**

### CODE REVIEW

File: internal/client/apidocs/apidocs.go [hunk 1]
```diff
@@ -150,8 +197,72 @@ func UpdateDocumentation(siteid string, id string, displayName string, openAPIDo
...
+	return apiclient.WriteByteArrayToFile(path.Join(folder, "apidocs.json"), false, respBody)
+}
+
+func Import(conn int, folder string) (err error) {
+	entities, err := readAPIDocsFile(path.Join(folder, "apidocs.json"))
+	if err != nil {
+		clilog.Error.Println("Error reading file: ", err)
+		return err
+	}
+	return fmt.Errorf("not implemented")
```
Comment (kurtkanaskie) [inline internal/client/apidocs/apidocs.go:257]: WIP?

**Review by kurtkanaskie — CHANGES_REQUESTED** [review]
Did you forget something?
```
func Import(conn int, folder string) (err error) {
	entities, err := readAPIDocsFile(path.Join(folder, "apidocs.json"))
	if err != nil {
		clilog.Error.Println("Error reading file: ", err)
		return err
	}
	return fmt.Errorf("not implemented")
}
```

File: internal/client/apidocs/apidocs.go (hunk 1 above)
Comment (srinandan) [reply internal/client/apidocs/apidocs.go]: Yes. But now I have added most if not all the functionality.

**Review by kurtkanaskie — CHANGES_REQUESTED** [review]
OK, tried the following.
Created 2 API docs in UI.

- Exported using --org and --siteid, all sites get exported, expected just the siteid I specified.
- Pruned the output files to be just those from the site.
- Deleted the API docs in the UI.
- Ran import.

```
gorun apidocs import --folder=apidocs-import --org=apigeex-mint-kurt --siteid=apigeex-mint-kurt-excotest
{
	"status": "success",
	"message": "API created",
	"requestId": "1620374756",
	"errorCode": "",
	"data": {
		"siteId": "apigeex-mint-kurt-excotest",
		"id": "448650",
		"title": "mint-riddles-v1",
		"description": "Riddles API ala carte",
		"published": true,
		"anonAllowed": true,
		"apiProductName": "mint-riddles-v1",
		"requireCallbackUrl": false,
		"imageUrl": "/files/riddles.jpeg?v=1701802032340",
		"categoryIds": [],
		"modified": "1702329207000",
		"visibility": true,
		"edgeAPIProductName": "mint-riddles-v1",
		"specId": "",
		"graphqlSchema": "",
		"graphqlEndpointUrl": "",
		"graphqlSchemaDisplayName": ""
	}
}

{
	"status": "success",
	"message": "API created",
	"requestId": "265270318",
	"errorCode": "",
	"data": {
		"siteId": "apigeex-mint-kurt-excotest",
		"id": "448651",
		"title": "mint-jokes-v1",
		"description": "Jokes API ala carte",
		"published": true,
		"anonAllowed": true,
		"apiProductName": "mint-jokes-v1",
		"requireCallbackUrl": false,
		"imageUrl": "/files/laugh.jpeg?v=1701801984172",
		"categoryIds": [],
		"modified": "1702329208000",
		"visibility": true,
		"edgeAPIProductName": "mint-jokes-v1",
		"specId": "",
		"graphqlSchema": "",
		"graphqlEndpointUrl": "",
		"graphqlSchemaDisplayName": ""
	}
}

Error: open apidocs_apigeex-mint-kurt-excotest_448640: no such file or directory
open apidocs_apigeex-mint-kurt-excotest_448641: no such file or directory
```
The API Docs where created but no documentation.
The error ids 448640 and 448641 are from the original export.

The import needs to use the IDs from the newly created API Docs and not those from the exported data.

**Review by kurtkanaskie — APPROVED** [review]

**Pull request merged**